src/expr.rs

```diff
@@ -504,6 +595,24 @@ impl<T: TensorType> Assign<T> {
     pub fn new_expr(variable: Expr<T>, value: Expr<T>) -> Expr<T> {
         Expr { expr: Rc::new(Assign::new(variable, value)) }
     }
+
+    /// Creates an expression that takes values from `iterable` to fill `variable`.
+    pub fn to(variable: Expr<T>, iterable: impl Iterator<Item = T>) -> ::Result<Expr<T>> {
+        let constant = if let ShapeHint::Exactly(shape) = variable.expr.shape_hint() {
+            let values: Vec<_> = iterable
+                .take(shape.iter().product::<u64>() as usize)
+                .collect();
+
+            Constant::new_expr(
+                Tensor::new(shape)
+                    .with_values(&values)?
+            )
+        } else {
+            panic!("Cannot assign to expression with unknown size!")
```

Comment:
Are we okay with panic'ing here? Is trying to assign to something that isn't variable something a user of the library would want to handle at runtime?

Reply:
If it's a condition that a user of the library can trigger, it should be an error, not a panic.